pkg/sql/txn_state.go

```diff
@@ -45,9 +45,11 @@ type txnState struct {
 	// Mutable fields accessed from goroutines not synchronized by this txn's
 	// session, such as when a SHOW SESSIONS statement is executed on another
 	// session.
+	//
 	// Note that reads of mu.txn from the session's main goroutine do not require
 	// acquiring a read lock - since only that goroutine will ever write to
-	// mu.txn.
+	// mu.txn. Writes to mu.txn do require a write lock to guarantee safety with
```

Comment:
I'd argue that this comment is not needed, but I guess I've forfeited that right.